python/gui/MaaS/cbv/DatasetFileWebsocketFilelike.py

```diff
@@ -0,0 +1,20 @@
+import asyncio
+from typing import AnyStr
+from dmod.client.request_clients import DatasetExternalClient
+
+
+class DatasetFileWebsocketFilelike:
+
+    def __init__(self, client: DatasetExternalClient, dataset_name: str, file_name: str):
+        self._client = client
+        self._dataset_name = dataset_name
+        self._file_name = file_name
+        self._read_index: int = 0
+
+    def read(self, blksize: int) -> AnyStr:
+
+        result = asyncio.get_event_loop().run_until_complete(
```

Comment:
do we want to try catch here? if the `download_item_block` call raises an exception and a downstream user of `DatasetFileWebsocketFilelike` catches an exception, the state of their `DatasetFileWebsocketFilelike` could be corrupted or atleast unclear what the behavior should be. Meaning, here, `_read_index`, if caught during a failed read, should continue from the failure point. Is this the expected behavior, or should _read_index be reset to `0`?